### PR TITLE
Add a list of subcategories to the Itunes Category

### PR DESCRIPTION
**Pull request by filaruina**
Necessary changes to other classes
Add a test related to subcategories

### CODE REVIEW

File: rome-modules/src/main/java/com/rometools/modules/itunes/io/ITunesParser.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public String getNamespaceUri() {
                     if (subcategory != null && subcategory.getAttribute("text") != null) {
                         final Subcategory subcat = new Subcategory();
                         subcat.setName(subcategory.getAttribute("text").getValue().trim());
-                        cat.setSubcategory(subcat);
+                        cat.addSubCategory(subcat);
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/itunes/io/ITunesParser.java]: Looks like it's only parsing the first subcategory unlike the generator which generates a list of subcategories?

File: rome-modules/src/main/java/com/rometools/modules/itunes/types/Category.java [hunk 2]
```diff
@@ -26,7 +29,7 @@
 
     private static final long serialVersionUID = 1L;
     private String name;
-    private Subcategory subcategory;
+    private List<Subcategory> subcategories;
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/itunes/types/Category.java]: Maybe initialize the list from the start?

File: rome-modules/src/main/java/com/rometools/modules/itunes/types/Category.java [hunk 3]
```diff
@@ -64,7 +67,20 @@ public void setName(final String name) {
      * @return Returns the Subcategory object for this category
      */
     public Subcategory getSubcategory() {
-        return subcategory;
+        return subcategories.get(0);
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/itunes/types/Category.java]: Backward incompatible change: if there are no subcategories it used to return null, now it's throwing null pointer.

File: rome-modules/src/main/java/com/rometools/modules/itunes/types/Category.java [hunk 4]
```diff
@@ -73,7 +89,21 @@ public Subcategory getSubcategory() {
...
     public void setSubcategory(final Subcategory subcategory) {
-        this.subcategory = subcategory;
+        subcategories = Lists.create(subcategory);
+    }
+
+    public void setSubcategories(final List<Subcategory> subcategories) {
+        this.subcategories = subcategories;
+    }
+
+    public void addSubCategory(final Subcategory subcategory) {
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/itunes/types/Category.java]: Inconsistent naming: `SubCategory` vs `Subcategory`

**Comment by mishako** [comment]
Left some comments. Other than that :+1:

Relevant quote from [help.apple.com](https://help.apple.com/itc/podcasts_connect/#/itcb54353390):

> Although you can specify more than one category and subcategory in your feed, Apple Podcasts only recognizes the first category and subcategory.

**Comment by filaruina** [comment]
@mishako Thanks for reviewing! I changed a few things and addressed your comments

**Pull request merged**